cassandra/src/main/scala/filodb.cassandra/Util.scala

```diff
@@ -84,4 +86,35 @@ object Util {
   def toBuffer(key: Array[Byte]): ByteBuffer = ByteBuffer.wrap(key)
 
   def toHex(bb: ByteBuffer): String = com.datastax.driver.core.utils.Bytes.toHexString(bb)
-}
\ No newline at end of file
+}
+
+/**
+ * Quick & Dirty synchronous exponential backoff that sleeps.
+ * Use only for synchronous API calls
+ */
+class RetryWithExpBackOffIterator(inner: Iterator[Row]) extends Iterator[Row] with StrictLogging{
+  val callStack = new RuntimeException()
+  override def hasNext: Boolean = inner.hasNext
+
+  //scalastyle:off null
+  override def next(): Row = {
+    var retries = 0
+    val maxRetries = 5 // Hardcode maxRetries for now. If needed make it a config later.
+    var nxt: Row = null
+    while (retries < maxRetries && nxt == null) {
+      try {
+        nxt = inner.next()
+      } catch {
+        case e: ReadTimeoutException =>
```

Comment:
We still have to think about UnavailableException in this context. This is a bit more tricky because we are dealing directly with the semantics of consistency and hiding behavior here may be a surprise to the user. Definitely should be bubbled up into a big, giant comment and in docs regardless.

I think for now, dealing with just the RTE is the easiest fix as that will avoid the common case of retrying when it just takes a little longer to churn through. 

For reference, there is a good explanation on the different types of exceptions on the Java-Driver's documentation about retry logic: https://github.com/datastax/java-driver/tree/3.x/manual/retries#onreadtimeout